src/include/processor/operator/persistent/reader/csv/parallel_csv_reader.h

```diff
@@ -37,15 +42,15 @@ struct ParallelCSVLocalState final : public function::TableFuncLocalState {
 
 struct ParallelCSVScanSharedState final : public function::ScanFileSharedState {
     explicit ParallelCSVScanSharedState(common::ReaderConfig readerConfig, uint64_t numRows,
-        uint64_t numColumns, main::ClientContext* context, common::CSVReaderConfig csvReaderConfig)
-        : ScanFileSharedState{std::move(readerConfig), numRows, context}, numColumns{numColumns},
-          numBlocksReadByFiles{0}, csvReaderConfig{std::move(csvReaderConfig)} {}
+        uint64_t numColumns, main::ClientContext* context, common::CSVReaderConfig csvReaderConfig);
 
     void setFileComplete(uint64_t completedFileIdx);
 
     uint64_t numColumns;
     uint64_t numBlocksReadByFiles = 0;
     common::CSVReaderConfig csvReaderConfig;
+    WarningCounter warningCounter;
+    std::vector<CSVErrorHandler> errorHandlers;
```

Comment:
I currently only have the error handlers in the shared state (no local error handlers) because even if I had local error handlers locking would still be needed when accessing the global warning counter (used to limit the # of global warnings).

This will still have additional performance penalty since the critical sections will be larger, but I kept it this way for simplicity. If we still want local CSV error handlers I'll do it in a separate PR since it isn't critical and this PR has already gotten quite bloated.

Reply:
If the warning limit only applies to each thread (it becomes a warning limit for each thread) implementing this would be simpler but I'm not sure how realistic that is

Reply:
We should optimize this in a separate PR under the case of a large waning limit. You can keep a small local cache for 2048 warnings, and merge once that's full.